### ISSUE TITLE
Achieve compatibility with Payload CMS

### ISSUE THREAD

**Issue by AlekSi** [FerretDB/FerretDB#2425]
https://github.com/payloadcms/payload
https://payloadcms.com

We should quickly check if it is compatible with FerretDB and write a blog post.

**Comment by TimHal** [comment]
Hi Alexey! I've heard about Ferret just a short time ago, thanks to Hacker news and was instantly thinking about payload. 
I've been using payload for quite a bit now and if I can help you/get involved I'd be happy to pick up any tasks and let you know how it goes, once I am able to set up everything.

The only thing that concerns me a bit is that Ferret does not support nested arrays, I can try to set up a minimal breaking example based on that over the weekend I hope.

**Comment by ptrfarkas** [comment]
@TimHal of course, you are welcome to get involved, and please let us know if we can help with anything here!

**Comment by b1ron** [comment]
So far all that's required for a payload-website is #3192. #1745 is also required but the application just sends an empty `fields` document we can do a bare bones implementation or even a stub.

**Comment by DanRibbens** [comment]
Hello from [Payload](https://github.com/payloadcms/payload)!

This is really cool to see you guys working on this!

> So far all that's required for a payload-website is #3192. #1745 is also required but the application just sends an empty `fields` document we can do a bare bones implementation or even a stub.

There is a chance that the use of `$first` goes away once we get our 2.0 changes finished. The branch and code to keep an eye on is https://github.com/payloadcms/payload/tree/feat/db-adapters/packages/db-mongodb.

I'll keep you posted if anything else changes.

**Comment by gwozdzm** [comment]
@DanRibbens and @TimHal - we recently released FerretDB 2.0 and all issues mentioned as blockers are supported now!

**Comment by TimHal** [comment]
Dear @gwozdzm , great news, glad to hear you are making good progress on this interesting project :)

I must admit that I have too much academic work on my hands these days and, after I left my last job, did not have time to keep up with payload as much as I wanted. For those reasons I can not really see myself writing the required adapters anytime soon. I hope that @DanRibbens or some other talented people from the community can pick this up, though! Sorry for the disappointing reply here, but I don't want to make commitments, which I can not follow through with :/

**Comment by Fashander** [comment]
All previous blockers #2128 #1745 are now supported in v2.x, and everything works as expected so far.
